### PR TITLE
drivers: regulator: enable driver when compatible node is defined 

### PR DESCRIPTION
**Pull request by jfischer-no**
Add regulator compatibility string to Kconfig.fixed file
and allow fixed regulator driver to be enabled when compatible
node is defined. However, remove this option from common Kconfig
as it is legitimate to enable/disable regulator support.

Change menuconfig REGULATOR_FIXED to config since this
menu nesting is not necessary.

### CODE REVIEW

File: drivers/regulator/CMakeLists.txt
Comment (pabigot) [inline drivers/regulator/CMakeLists.txt:4]: Every other driver subsystem puts its sources into a library.  This one should too when another type of regulator is added.  So are we just kicking the problem down the road?

Also, the next line adds `regulator_fixed.c` as a library source.  What library is it going into?

Comment (jfischer-no) [reply drivers/regulator/CMakeLists.txt:4]: > Every other driver subsystem puts its sources into a library.

No, this is not consistent. I will think it over again.


> Also, the next line adds `regulator_fixed.c` as a library source. What library is it going into?

my mistake, this should probably be `zephyr_sources_ifdef(CONFIG_REGULATOR_FIXED regulator_fixed.c`) in that case.

Comment (carlescufi) [reply drivers/regulator/CMakeLists.txt:4]: @tejlmand the fact that enabling `CONFIG_REGULATOR` in `tests/drivers/regulator/fixed/` makes that test fail for boards that do not have a regulator in their Devicetree is what triggers the problem with `zephyr_library()`.

Comment (tejlmand) [reply drivers/regulator/CMakeLists.txt:4]: > Every other driver subsystem puts its sources into a library. 

Unfortunately not, this is very inconsistent among Zephyr driver libraries.

One root cause is probable that driver libs in general don't use depends (ask someone else why they don't as I don't know) but the sources added to the driver lib uses `zephyr_[library]_sources_ifdef()` which means that if `zephyr_library()` was used, then it could end up being empty and thus result in CMake failures.

And thus people take the easy but dirty solution of just placing it `zephyr_sources()` resulting in this mess instead.
https://github.com/zephyrproject-rtos/zephyr/issues/8825

Comment (tejlmand) [reply drivers/regulator/CMakeLists.txt:4]: @jfischer-no I would prefer we keep `zephyr_library()` to avoid making more mess as described in #8825.

Looking into how this can be accomplished.

Comment (jfischer-no) [reply drivers/regulator/CMakeLists.txt]: I agree on zephyr_library(). I reverted this changes and add comment to tests/drivers/regulator/fixed/prj.conf

File: tests/drivers/regulator/fixed/prj.conf
Comment (tejlmand) [inline tests/drivers/regulator/fixed/prj.conf:4]: please check: https://github.com/zephyrproject-rtos/zephyr/pull/34289
for a more general fix that covers the empty Zephyr lib in general for drivers.

**Review by pabigot — APPROVED** [review]
Seems fine; hope somebody takes up this driver API and gets the supporting dependencies in place so it can be used.

**Comment by jfischer-no** [comment]
Wait until #34289 is merged.

File: tests/drivers/regulator/fixed/testcase.yaml
Comment (emob-nordic) [inline tests/drivers/regulator/fixed/testcase.yaml:9]: Not thingy52?

Comment (emob-nordic) [reply tests/drivers/regulator/fixed/testcase.yaml:9]: Could be both i suppose.

Comment (jfischer-no) [reply tests/drivers/regulator/fixed/testcase.yaml:9]: Not Thingy52. Thingy's do not have arduino gpio headers.

Comment (emob-nordic) [reply tests/drivers/regulator/fixed/testcase.yaml:9]: Ah yes, sorry i was thinking in terms of regualtors.

**Review by eobalski — APPROVED** [review]

File: tests/drivers/regulator/fixed/prj.conf
Comment (jfischer-no) [reply tests/drivers/regulator/fixed/prj.conf]: rebased and removed comment

Comment (tejlmand) [reply tests/drivers/regulator/fixed/prj.conf]: @jfischer-no #34289 has been merged, please update the prj.conf to not contain workarounds, so we can see if there are any other outstanding issues in the build system.

**Review by tejlmand — APPROVED** [review]

**Pull request merged**
